app/services/change_offer.rb

```diff
@@ -21,9 +21,10 @@ def save!
         ActiveRecord::Base.transaction do
           ApplicationStateChange.new(application_choice).make_offer!
 
+          persist_conditions_service.save
```

Comment:
As mentioned in the description, this is behaving a bit like an Offer model. We pass in an unsaved offer to ChangeOffer. Then pass it into the validator and save it if it's valid inside ChangeOffer.

Any thoughts about patterns to use here?

Reply:
I'm thinking this because we have two public methods on the service object, which breaks the usual implementation of a service object